src/ext/java/org/opentripplanner/ext/siri/SiriTimetableSnapshotSource.java

```diff
@@ -373,11 +376,10 @@ private Result<UpdateSuccess, UpdateError> addTripToGraphAndBuffer(TripUpdate tr
       serviceDate
     );
 
-    // Add new trip times to the buffer and return success
+    // Add new trip times to the buffer and return result with success or error. The update method
+    // will perform protective copies as needed, whether pattern is created by realtime data or not.
```

Comment:
From a maintenance perspective, it can be very helpful to have an inline narrative of what a method is doing including some non-obvious details, without needing to read and piece together the details of each called method's own documentation. I generally added comments like this for details that tripped me up, which I thought could help future comprehension or refactoring. 

I have moved the new detail about the behavior of the called method into that method's Javadoc, while keeping a one-line note here in the caller.